### PR TITLE
[Console] Use secrets manager password for metadata & snapshot

### PR DESCRIPTION
**Pull request by mikaylathompson**
### Description
While #857 added support for password-containing-secrets in CDK and when making cluster commands via the Console Library, it did not support them for creating snapshots and migrating metadata. This PR updates the library to pass those through to those commands.

Specifically, the console library still pulls the secret and then passes the password itself to the java command.

It also masks the password in log messages.

### Issues Resolved
Partially [MIGRATIONS-1912](https://opensearch.atlassian.net/browse/MIGRATIONS-1912) (this also needs sigv4 to be complete), I don't think we have a task for the metadata side.

### Testing
Unit & manual

### Check List
- [x] New functionality includes testing
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/metadata.py
Comment (AndreKurait) [inline TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/metadata.py:181]: nit: in the future i can see metadata migrate needing access to source cluster in which case password_field can be renamed sensitive fields and be a list

File: TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/metadata.py
Comment (AndreKurait) [inline TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/metadata.py:142]: nit: Should we be converting all the values to string at the end instead?

File: TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/tests/test_command_runner.py
Comment (AndreKurait) [inline TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/tests/test_command_runner.py:1]: Can we test/support argument flags which don't have a value? e.g. `--debug`

How will this be represented in the dict? `{ "--debug" : ""}` or `{"--debug": {}}` or something else? Is there a different data structure which may make this more intuitive?

Comment (mikaylathompson) [reply TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/tests/test_command_runner.py:1]: Yeah, I did it with `None`. Maybe I make a `FlagOnly` type that can passed through. I want the dictionary in large part because it makes it much easier to figure out where the password might be.

File: TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/command_runner.py
Comment (AndreKurait) [inline TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/command_runner.py:37]: nit: should we return command.copy() to be consistent here?

File: TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/command_runner.py
Comment (AndreKurait) [inline TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/command_runner.py:42]: This is a bit unintuitive, we're taking the fields as inputs, but looking for the value and replacing it (instead of reconstructing the command_args dict and reconstructing the display command).

There's a slight bug in this logic in that we're sensitizing the first value that we find. e.g.

Given args: `--username admin --password admin`
Our Display_Command will be `--username ******** --password admin`

**Review by AndreKurait — CHANGES_REQUESTED** [review]

File: TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/command_runner.py
Comment (AndreKurait) [inline TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/command_runner.py:38]: nit: replace password with sensitive_value in this function

File: TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/command_runner.py
Comment (mikaylathompson) [reply TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/command_runner.py:42]: Oh, great point. I'll add that as a test and rework it.

File: TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/command_runner.py
Comment (AndreKurait) [inline TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/command_runner.py:24]: nit: do we need to add \"? Can the values contain spaces? Do we have a test that tests for this?

password: my password

File: TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/command_runner.py
Comment (AndreKurait) [inline TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/console_link/models/command_runner.py:41]: nit: I'm still a bit hesitant on this logic and would rather see us use command_args for this. It's unlikely to cause a bug, but the possibility is still there to log something sensitive if another bug existed in the system.

E.g. 1:
if command_root == sensitive_field, there will be an issue (shouldn't happen)

e.g 2:
If value == sensitive field, there will be an issue (a lot less likely than before, but could still happen due to other issue.

Say someone accidentally sets their username as "--target-password", then password will be logged in plain text.

**Review by AndreKurait — APPROVED** [review]
Please don't merge this until after #866

**Pull request merged**
